cmd/main/main.go

```diff
@@ -78,17 +79,31 @@ func main() {
 			Password: os.Getenv("ST_DB_PASSWORD"),
 		},
 	}
+	log.Printf("debug: configuration: %+v", conf)
 
 	db, err := db.New(conf.db)
 	if err != nil {
 		log.Fatalf("unable to create database: %+v", err)
 	}
-	schemaVersion, err := schema.GetCurrentSchemaVersion(db)
+
+	goose.SetBaseFS(migrations.Migrations)
+	autoInit, _ := strconv.ParseBool(os.Getenv("ST_DB_AUTO_INIT"))
+	if autoInit {
+		log.Printf("auto-initialising database")
+		if err := goose.SetDialect("postgres"); err != nil {
+			log.Fatalf("failed to set goose dialect: %+v", err)
+		}
+		if err := goose.Up(db.DB, "."); err != nil {
+			log.Fatalf("failed to run goose migrations: %+v", err)
+		}
+	}
+
+	upToDate, err := migrations.IsUpToDate(db.DB)
 	if err != nil {
-		log.Fatalf("unable to get schema version: %+v", err)
+		log.Fatalf("failed to check if migrations are up to date: %+v", err)
 	}
-	if schemaVersion != schema.HighestSchemaVersion {
-		log.Fatalf("Database schema outdated, please run 'init' again; expected %d, but found %d", schema.HighestSchemaVersion, schemaVersion)
+	if !upToDate {
+		log.Fatalf("database not up to date, please run 'init' or set ST_DB_AUTO_INIT=true")
```

Comment:
Is it worth moving this to `db.New()`?

Reply:
Problem is that `cmd/init` also uses `db.New()`, but there we want to skip the outdated check - do you think it's worth adding a flag to `db.New()` for that?

Reply:
Could make it an option in `db.Config`?